src/mapblock.h

```diff
@@ -639,26 +639,38 @@ typedef std::vector<MapBlock*> MapBlockVect;
 
 inline bool objectpos_over_limit(v3f p)
 {
-	const static float map_gen_limit_bs = MYMIN(MAX_MAP_GENERATION_LIMIT,
-		g_settings->getU16("map_generation_limit")) * BS;
-	return (p.X < -map_gen_limit_bs
-		|| p.X >  map_gen_limit_bs
-		|| p.Y < -map_gen_limit_bs
-		|| p.Y >  map_gen_limit_bs
-		|| p.Z < -map_gen_limit_bs
-		|| p.Z >  map_gen_limit_bs);
+	const static u16 max_map_gen_limit = MYMIN(MAX_MAP_GENERATION_LIMIT,
+		g_settings->getU16("map_generation_limit"));
+	const static float map_gen_limit_x_bs = MYMIN(max_map_gen_limit,
+		g_settings->getU16("map_generation_limit_x")) * BS;
+	const static float map_gen_limit_y_bs = MYMIN(max_map_gen_limit,
+		g_settings->getU16("map_generation_limit_y")) * BS;
+	const static float map_gen_limit_z_bs = MYMIN(max_map_gen_limit,
+		g_settings->getU16("map_generation_limit_z")) * BS;
+	return (p.X < -map_gen_limit_x_bs
```

Comment:
There is no need to surround this in parenthesis. return is a keyword, not a function.